nixos/doc/manual/development/reviewing-contributions.xml

```diff
@@ -0,0 +1,257 @@
+
+<para>When reviewing a pull request, please always be nice and polite. Controversial changes can lead to controversial opinions, but it is important to respect every community members and their work.</para>
+
+<para>Github provides emoji, they are a simple and quick way to provide feedback to pull-requests or any comments. The thumb-down emoji should be used with care and if possible accompanied with some explanations so the submitter has directions to improve his contribution.</para>
+
+<para>Pull-requests reviews should include a list of what has been reviewed in a comment, so other reviewers and mergers can know the state of the review.</para>
+
+<para>All the review template samples that are provided in this section are generic, and should be adapted by the reviewer to fit the pull-request content by adding or removing review points.</para>
+
+<section><title>Package updates</title>
+
+<para>A package update is the most trivial and common type of pull-request. These pull-requests mainly consist in updating the version part of the package name and the source hash.</para>
+<para>It can happen that non trivial updates include patches or more complex changes.</para>
+
+<para>Reviewing process:</para>
+
+<itemizedlist>
+  <listitem><para>Add labels to the pull-request.</para>
+    <itemizedlist>
+      <listitem><para><literal>8.has: package (update)</literal> and any topic label that fit the updated package.</para></listitem>
+    </itemizedlist>
+  </listitem>
+  <listitem><para>Checking that the package versioning is fitting the guidelines.</para></listitem>
+  <listitem><para>Checking that the commit text is fitting the guidelines.</para></listitem>
+  <listitem><para>Checking that the package maintainers are notified.</para>
+    <itemizedlist>
+      <listitem><para>mention-bot usually notify github users based on the submitted changes, but it can happen that it miss some of the package maintainers.</para></listitem>
+    </itemizedlist>
+  </listitem>
+  <listitem><para>Checking that the code contains no typos.</para></listitem>
+  <listitem><para>Building the package locally.</para>
+    <itemizedlist>
+      <listitem><para>Pull-requests are often targetted to the master or staging branch so building the pull-request locally as it is submitted can trigger a large amount of source builds.</para>
+        <para>It is possible to rebase the changes on nixos-unstable or nixpkgs-unstable for easier review.</para>
+<screen>
+TODO: add commands for a local rebase and build
+</screen>
+      </listitem>
+    </itemizedlist>
+  </listitem>
+  <listitem><para>Running every binary.</para></listitem>
+  <listitem><para>Checking that the packages using the updated package are building fine.</para>
+<screen>
+TODO: add commands
```

Comment:
Actually it's called the `installCheckPhase`. It's not widely used but here is an example: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/zotero/default.nix#L72-L73
the `checkPhase` is run just after the build (typically some `make test`) and the `installCheckPhase` is run after install